bin/zowe-verify-authenticity.sh

```diff
@@ -0,0 +1,374 @@
+#!/bin/sh
+#######################################################################
+# This program and the accompanying materials are made available
+# under the terms of the Eclipse Public License v2.0 which
+# accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright Contributors to the Zowe Project. 2020
+#######################################################################
+
+#	zowe-verify-authenticity.sh
+
+# Function: Verify the authenticity of a Zowe runtime driver
+# Method  : Create a hash of every file in the driver and compare them 
+#           to the hashes of the official version.  
+# Inputs  -	pathname of Zowe runtime driver on your z/OS system
+#         -	HashFiles.class (binary)
+#         -	RefRuntimeHash-v.r.m.txt (list of files with hash keys)
+#  
+# SMP/E   - The directory SMPE in the runtime folder is excluded from this check
+#           The SMPE directory and contents must not be present in the supplied
+#           RefRuntimeHash-v.r.m.txt file, or they will be flagged here as missing
+#
+# fingerprint
+#         - The 'fingerprint' directory in the runtime directory holds 
+#           the reference hash key file RefRuntimeHash-v.r.m.txt
+#           Like the SMPE directory, it's excluded from the check.
+#
+# Outputs - Lists of runtime files missing, extra 
+#           and different from official Ref version
+#           
+# Requires- java runtime
+
+SCRIPT=zowe-verify-authenticity
+echo $SCRIPT.sh started
+notD="does not exist or is not a directory"
+
+while getopts "h:r:f:l:" opt; do
+  case $opt in
+    r) runtimePath=$OPTARG
+        if [[ ! -d $runtimePath ]]
+        then
+            echo Error: -r $runtimePath $notD
+            exit 1
+        fi 
+        runtimePath=$(cd $runtimePath;pwd)
+        ;;
+    h) hashPath=$OPTARG
+        if [[ ! -d $hashPath ]]
+        then
+            echo Error: -h $hashPath $notD
+            exit 1
+        fi    
+        hashPath=$(cd $hashPath;pwd)
+        ;;
+    f) refPath=$OPTARG
+        if [[ ! -d $refPath ]]
+        then
+            echo Error: -f $refPath $notD
+            exit 1
+        fi
+        refPath=$(cd $refPath;pwd)
```

Comment:
These checks are for what the user specified.  The later checks are for the defaults.

Reply:
I understand the code, but the duplicated logic is what I was raising

Reply:
Lines at 43 onward check a directory, issue a message and assign a value.  Lines at 90 onward check a different directory, issue a different message and assign a different value.  Yes, it's the same logic, but with different data.  Do you see a simple way to avoid the duplication?